python/python/tests/test_dataset.py

```diff
@@ -269,8 +270,18 @@ def adder(batch: pa.RecordBatch) -> pa.RecordBatch:
 
     fragment = fragment.add_columns(adder, columns=["a"])
     schema = dataset.schema.append(pa.field("c", pa.int64()))
-    dataset = dataset._create_version_from_fragments(schema, [fragment])
+    dataset = lance.LanceDataset._commit(base_dir, schema, [fragment])
     tbl = dataset.to_table()
     assert tbl == pa.Table.from_pydict(
         {"a": range(100), "b": range(100), "c": pa.array(range(0, 200, 2), pa.int64())}
     )
+
+def test_create_from_fragments(tmp_path: Path):
+    table = pa.Table.from_pydict({"a": range(100), "b": range(100)})
+    base_dir = tmp_path / "test"
+    fragment = lance.fragment.LanceFragment.create(base_dir, 1, table)
```

Comment:
would the input data to create ever be a LanceDataset? I'm wondering if this has the same tokio runtime issue that the read/write APIs have. If so, you may need to convert the input LanceDataset into a pyarrow Table first until we figure out how to deal with that.

Reply:
at fragment level? seems no. If you consider this is the first step to create fragment. this is usually just write in memory data to disk.